### DISCUSSION TITLE
Custom GraphQL query prefixing an underscore to ctx.param keys

### DISCUSSION THREAD

**Discussion by derrickmehaffy** [strapi/strapi#6510] [Legacy Topics]
While working with a user on twitch to help with the lifecycle callbacks, we created a custom controller to query by a specific field and the question came up about GraphQL as well.

Here is the repo we were testing with: https://github.com/derrickmehaffy/testing-strapi-lifecycles

With REST the following works fine (controller):

```js
const { sanitizeEntity } = require('strapi-utils');

module.exports = {

  async findOneByUID(ctx) {

    let { uid } = ctx.params;

    const entity = await strapi.query('blog-post').findOne({ uid: uid });
    return sanitizeEntity(entity, { model: strapi.models['blog-post'] });
  },
};
```

However when we go to customize the [schema](https://github.com/derrickmehaffy/testing-strapi-lifecycles/blob/master/api/blog-post/config/schema.graphql.js): 

```js
module.exports = {
  query: `
    blogPostByUID(uid: String!): BlogPost
  `,
  resolver: {
    Query: {
      blogPostByUID: {
        description: 'Return a single blog post by uid',
        resolver: 'application::blog-post.blog-post.findOneByUID'
      }
    },
  },
};
```

And try to run the GQL query it's prefixing the `uid` with an underscore like `_uid`, this seems odd to me as the default `findOne` query in the GQL schema doesn't do this with `id`.

Is this intended (I dove through the GQL docs, which I really think need to be rewritten) and could not find any references to this or why it happens.

As a work around we were able to just check for the `_uid` (although ugly):

```js
  async findOneByUID(ctx) {

    // Using GraphQL this returns a _uid but with REST it returns uid????
    console.log(ctx.params)

    let { uid } = ctx.params;

    // Work around for GQL _uid
    if (ctx.params._uid) {
      uid = ctx.params._uid
    }

    const entity = await strapi.query('blog-post').findOne({ uid: uid });
    return sanitizeEntity(entity, { model: strapi.models['blog-post'] });
  },
```

**Comment by russmilburn** [comment]
I have also come across this issue

**Reply by russmilburn** [reply]
hey @derrickmehaffy 

i have figured out a work around if you add a resolverOf and a custom method to resolver to your schema.  You can pass an options object. the option object will contain your augments without the underscore

resolverOf: 'application::blog-post.blog-post.findOneByUID',
        resolver: async (obj, options, ctx) => {
          return await strapi.controllers.blog-post.findOneByUID(ctx, options);
}
